src/main.rs

```diff
@@ -49,139 +49,198 @@ fn run() -> Result<(), failure::Error> {
                         )
                         .arg(
                             Arg::with_name("file")
+                            .help("The value passed in is a filename; open and upload its contents")
                             .short("f")
                             .long("file")
                             .takes_value(false)
-                            .help("the value passed in is a filename; open and upload its contents"),
                         )
                 )
                 .subcommand(
-                    SubCommand::with_name("write-bulk")
-                        .about("upload multiple key-value pairs at once")
+                    SubCommand::with_name("get")
+                        .about("Get a key's value from a namespace")
                         .arg(
                             Arg::with_name("namespace-id")
-                                .help("the id of your Workers KV namespace")
-                                .required(true)
-                                .index(1),
+                            .help("The ID of the namespace this action applies to")
+                            .required(true)
+                            // .short("n")
+                            // .long("namespace-id")
+                            // .value_name("<ID>")
+                            // .takes_value(true)
                         )
                         .arg(
-                            Arg::with_name("path")
-                            .help("the json file of key-value pairs to upload, in form [{\"key\":..., \"value\":...}\"...].")
+                            Arg::with_name("key")
+                            .help("Key whose value to get")
                             .required(true)
-                            .index(2),
                         )
                 )
                 .subcommand(
-                    SubCommand::with_name("delete-key")
+                    SubCommand::with_name("delete")
+                        .about("Delete a key (and its value) from a namespace")
                         .arg(
                             Arg::with_name("namespace-id")
+                            .help("The ID of the namespace this action applies to")
                             .required(true)
+                            // .short("n")
+                            // .long("namespace-id")
+                            // .value_name("<ID>")
+                            // .takes_value(true)
                         )
                         .arg(
                             Arg::with_name("key")
+                            .help("Key whose value to delete")
                             .required(true)
                         )
                 )
                 .subcommand(
-                    SubCommand::with_name("delete-bulk")
+                    SubCommand::with_name("list")
+                        .about("List all keys in a namespace. Produces JSON output")
```

Comment:
should we also print namespace ids as json to start and introduce table printing for both later as a flag? like `--table`?

Reply:
We could! I still am hesitant about printing tables, given the a) dynamic number of items in each entry and b) how big this output is going to get. I think folks, at most, would want a csv/tsv print format on top of json printing.